lib/autodiscover/client.rb

```diff
@@ -10,10 +10,11 @@ class Client
     #   with something other than the e-mail. For instance DOMAIN\user
     # @param domain [String] An optional domain to provide as an override for
     #   the one parsed from the e-mail.
-    def initialize(email:, password:, username: nil, domain: nil)
+    def initialize(email:, password:, username: nil, domain: nil, connect_timeout: nil)
```

Comment:
I think it probably makes sense to  have a reasonable default here. How does 10 seconds sound? If you could make it a constant for the `Client` class like `DEFAULT_HTTP_TIMEOUT` and then pass it in the params I'll merge the code and cut a new gem.